aggregator/mock/aggregator.go

```diff
@@ -107,3 +112,30 @@ func (agg *mockAggregator) Snapshot() SnapshotResult {
 
 	return result
 }
+
+func cloneMetric(m unaggregated.MetricUnion) unaggregated.MetricUnion {
+	mu := m
+	if !m.OwnsID {
```

Comment:
The meaning of `OwnsID` is that if it's true, whoever consumes the id takes the ownership of the id and it is guaranteed that the id cannot be mutated externally, and if it's false, whoever consumes the id needs to make a copy of the id (since the id is not owned by the union and can then be mutated by the owner).